pythonforandroid/recipe.py

```diff
@@ -725,14 +725,44 @@ class PythonRecipe(Recipe):
     This is almost always what you want to do.'''
 
     setup_extra_args = []
-    '''List of extra arugments to pass to setup.py'''
+    '''List of extra arguments to pass to setup.py'''
+
+    depends = [('python2', 'python3')]
+    '''
+    .. note:: it's important to keep this depends as a class attribute, outside
+              `__init__` because, sometimes, we only initialize the object, so
+              the `__init__` call it won't be called, which will lead to not
+              have the python versions as a dependencies and it will cause a
+              tremendous `test_graph` error (difficult to track) and also, the
+              build order for dependencies will not be computed as expected (if
+              computed...). So be very careful with this line!!
+
+    .. warning:: this `depends` may be overwrote in inherited classes of
+                 `PythonRecipe`, so we make sure that any sub class will
+                 contain python as a dependency. We do this by checking the
+                 dependencies in meth:`PythonRecipe.__init__` method and adding
+                 them again in case that is necessary, so don't forget to call
+                 `super` in any inherited class of this class.
+    '''
 
     def __init__(self, *args, **kwargs):
         super(PythonRecipe, self).__init__(*args, **kwargs)
-        depends = self.depends
-        depends.append(('python2', 'python3', 'python3crystax'))
-        depends = list(set(depends))
-        self.depends = depends
+        if not any(
+            [
+                d
+                for d in {'python2', 'python3', ('python2', 'python3')}
+                if d in self.depends
+            ]
+        ):
+            # we overwrote `depends` in inherited recipe, so we must add it
```

Comment:
I would repharse this comment to:

```
# We ensure here that the recipe depends on python even it overrode `depends`.
# We only do this if it doesn't already depend on any python,
# since some recipes intentionally don't depend on/work with all python variants
```